run.sh

```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env sh
 set -e
 
-cd "${0%/*}"
+cd "$(dirname "$0")"
```

Comment:
Is everything fine with escaping here? I see quotes inside quotes.

Reply:
Yes, the idea is that the thing inside the `$(...)` is valid command line.

It is documented in a little bit of convoluted way in https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_02_03 and https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_03 .

To check it you can use the following simple shell script:
```sh
set -e
set -x

MSG='-e foo bar\n \c baz'

echo "unescaped (baz not shown):"
echo $MSG
echo "still unescaped (baz not shown):"
echo "$(echo $MSG)"
echo "escaped (baz shown):"
echo "$(echo "$MSG")"
```

P.S. As a side note, you could also use `readlink -f "$0"` to handle the case where `run.sh` is being symlinked (`$0` will the symlink name and `cd` without `readlink` would change to symlink's dir), but I decided against it.

Reply:
Ok. If you tested it I am merging.